share/dkg/pedersen/dkg.go

```diff
@@ -121,6 +121,43 @@ func NewDistKeyGenerator(suite Suite, longterm kyber.Scalar, participants []kybe
 	}, nil
 }
 
+
+func ResharingDKG(suite Suite, longterm kyber.Scalar, participants []kyber.Point, t int) (*DistKeyGenerator, error){
```

Comment:
I would prefer to have one `NewDistKeyGenerator` and one `NewDistKeyGeneratorWithSecret` (for example) so `NewDistKeyGenerator` only calls `NewDistKeyGeneratorWithSecret` with a random secret, so the code is not repeated.
Plus that will serve as a basis for future extensions like increasing/decreasing threshold etc..